python/ray/ml/predictors/integrations/sklearn/sklearn_predictor.py

```diff
@@ -36,12 +45,24 @@ def from_checkpoint(cls, checkpoint: Checkpoint) -> "SklearnPredictor":
                 ``SklearnTrainer`` run.
 
         """
-        raise NotImplementedError
+        path = checkpoint.to_directory()
+        estimator_path = os.path.join(path, MODEL_KEY)
+        with open(estimator_path, "rb") as f:
+            estimator = cpickle.load(f)
+        preprocessor_path = os.path.join(path, PREPROCESSOR_KEY)
+        if os.path.exists(preprocessor_path):
+            with open(preprocessor_path, "rb") as f:
+                preprocessor = cpickle.load(f)
+        else:
+            preprocessor = None
+        shutil.rmtree(path)
```

Comment:
I was going to suggest that as well! Would be quite useful. `to_directory` could also take in an argument to avoid making a temp copy if the checkpoint is already a directory.

Reply:
`with checkpoint.to_temp_directory() as tempdir` sounds useful.

I wouldn't go with the argument here - while it is nice to avoid inspecting internal state (e.g. in HF trainer checking the `_local_path` attribute) we don't know if we should clean up the returned path or not as we don't know if it's a temporary path or the existing (potentially durable) path. 

Actually, `with checkpoint.to_temp_directory() as tempdir` could just return the durable path if it's already a directory checkpoint. Probably should have a different name then, though.

Reply:
Suggested as `Checkpoint.as_directory()`: https://github.com/ray-project/ray/pull/23908/files